pkg/cmd/config/options.go

```diff
@@ -0,0 +1,174 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/component-base/cli/flag"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+	"github.com/gardener/gardenctl-v2/pkg/config"
+)
+
+// Options is a struct to support all config subcommands
+type options struct {
+	base.Options
+	// Configuration is the gardenctl configuration
+	Configuration *config.Config
+	// Command is the name of the config command
+	Command string
+	// Name is a unique name of this Garden that can be used to target this Garden
+	Name string
+	// KubeconfigFlag is the path to the kubeconfig file of the Garden cluster
+	KubeconfigFlag flag.StringFlag
+	// ContextFlag Overrides the current-context of the garden cluster kubeconfig
+	// +optional
+	ContextFlag flag.StringFlag
+	// Patterns is a list of regex patterns that can be defined to use custom input formats for targeting
+	// Use named capturing groups to match target values.
+	// Supported capturing groups: project, namespace, shoot
+	// +optional
+	Patterns []string
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *options) Complete(f util.Factory, cmd *cobra.Command, args []string) error {
+	manager, err := f.Manager()
+	if err != nil {
+		return fmt.Errorf("failed to get target manager: %w", err)
+	}
+
+	config := manager.Configuration()
+	if config == nil {
+		return errors.New("failed to get configuration")
+	}
+
+	o.Configuration = config
+
+	switch o.Command {
+	case "set-garden", "delete-garden":
+		if len(args) > 0 {
+			o.Name = strings.TrimSpace(args[0])
+		}
+	case "view":
+		if o.Output == "" {
+			o.Output = "yaml"
+		}
+	}
+
+	return nil
+}
+
+// Validate validates the provided options
+func (o *options) Validate() error {
+	switch o.Command {
+	case "set-garden", "delete-garden":
+		if o.Name == "" {
+			return errors.New("garden identity is required")
+		}
+	case "view":
+		return o.Options.Validate()
+	}
+
+	return nil
+}
+
+// AddFlags adds flags to adjust the output to a cobra command
+func (o *options) AddFlags(flags *pflag.FlagSet) {
+	switch o.Command {
```

Comment:
see my comment above. I'm also a bit unsure if this is a good idea in this case. However there may be cases (e.g. when we refactor the target command) where these approach can help to make the code cleaner, because we have a lot of overlap of the subcommands. But not in this case.

Reply:
I have split all subcommands into single files.